jcl/src/java.base/share/classes/jdk/internal/misc/Unsafe.java

```diff
@@ -0,0 +1,6140 @@
+	 * @param offset position of the value in obj
+	 * @return long value stored in obj
+	 */
+	public native long getLongVolatile(Object obj, long offset);
+
+	/**
+	 * Atomically sets the value of the long in the obj parameter at memory offset.
+	 * 
+	 * @param obj object into which to store the value
+	 * @param offset position of the value in obj
+	 * @param value long to store in obj
+	 */
+	public native void putLongVolatile(Object obj, long offset, long value);
+
+	/**
+	 * Atomically gets the value of the float in the obj parameter referenced by offset.
+	 * 
+	 * @param obj object from which to retrieve the value
+	 * @param offset position of the value in obj
+	 * @return float value stored in obj
+	 */
+	public native float getFloatVolatile(Object obj, long offset);
+
+	/**
+	 * Atomically sets the value of the float in the obj parameter at memory offset.
+	 * 
+	 * @param obj object into which to store the value
+	 * @param offset position of the value in obj
+	 * @param value float to store in obj
+	 */
+	public native void putFloatVolatile(Object obj, long offset, float value);
+
+	/**
+	 * Atomically gets the value of the double in the obj parameter referenced by offset.
+	 * 
+	 * @param obj object from which to retrieve the value
+	 * @param offset position of the value in obj
+	 * @return double value stored in obj
+	 */
+	public native double getDoubleVolatile(Object obj, long offset);
+
+	/**
+	 * Atomically sets the value of the double in the obj parameter at memory offset.
+	 * 
+	 * @param obj object into which to store the value
+	 * @param offset position of the value in obj
+	 * @param value double to store in obj
+	 */
+	public native void putDoubleVolatile(Object obj, long offset, double value);
+
+	/**
+	 * Makes permit available for thread parameter.
+	 * 
+	 * @param thread the thread to unpark. If null, method has no effect
+	 */
+	public native void unpark(Object thread);
+
+	/**
+	 * Disables current thread unless permit is available. Thread will not be 
+	 * scheduled until unpark provides permit.
+	 * 
+	 * @param isAbsolute if true park timeout should be absolute
+	 * @param time parks for time in ns. 
+	 *	 ms is expected if isAbsolute is set to true.
+	 */
+	public native void park(boolean isAbsolute, long time);
+
+	/**
+	 * Inserts an acquire memory fence, ensuring that no loads before this fence
+	 * are reordered with any loads/stores after the fence.
+	 */
+	public native void loadFence();
+
+	/**
+	 * Inserts a release memory fence, ensuring that no stores before this fence
+	 *  are reordered with any loads/stores after the fence.
+	 */
+	public native void storeFence();
+
+	/**
+	 * Inserts a complete memory fence, ensuring that no loads/stores before this
+	 *  fence are reordered with any loads/stores after the fence.
+	 */
+	public native void fullFence();
+	
+	/* 
+	 * sun.misc.Unsafe natives 
+	 */
+	
+	/* 
+	 * Allocates a block and adds it to a doubly-linked list for tracking purposes.  Actual 
```

Comment:
`doubly-linked list for tracking purposes` is that an OpenJDK or OpenJ9 statement?

Reply:
I took this from OpenJ9 `runtime/jcl/common/unsafe_mem.c:unsafeAllocateMemory`